vantiq-cloud-infra-operations/docs/eng/kubernetes-upgrade.md

```diff
@@ -0,0 +1,206 @@
+
+
+#### Upgrade with AWS Management Console
+
+1. Specify the cluster in Amazon EKS and execute the Control Plane upgrade with **Update Now**.  
+![eks1](../../imgs/kubernetes-upgrade/eks-1.png)
+
+1. Once the Control Plane upgrade is complete, select the Nodepool from the Node Group and execute the Node Pool upgrade with **Update Now**. Repeat for as many Node Group as necessary and upgrade the versions of each.  
+![eks2](../../imgs/kubernetes-upgrade/eks-2.png)
+
+
+#### Upgrade with Terraform
+
+The following is the procedure assuming that it was installed using the template ["AWS EKS for Vantiq with Terraform"](../../../vantiq-cloud-infra-operations/terraform_aws/readme_en.md). If it used a different template, make the necessary changes accordingly.  
+
+1. Change the value of `cluster_version` in `main.tf` to the version of the target.  
+
+```terraform
+module "eks" {
+  source             = "../modules/eks"
+  cluster_name       = local.cluster_name
+  env_name           = local.env_name
+  vpc_id             = module.vpc.vpc_id
+  public_subnet_ids  = module.vpc.public_subnet_ids
+  private_subnet_ids = module.vpc.private_subnet_ids
+
+  worker_access_ssh_key_name = local.worker_access_ssh_key_name
+  basion_ec2_sg_ids          = [aws_security_group.basion-ssh-allow.id]
+
+  keycloak_db_expose_port = local.keycloak_db_expose_port
+  keycloak_db_sg_id       = module.keycloak-db.keycloak_db_sg_id
+
+  # The following is custom setting
+  cluster_version = "1.21"  # 1.20 -> 1.21　
+```
+
+2. Execute the `terraform plan` and verify that the changes are correct. If any unexpected changes are detected at this point, resolve them first.  
+
+3. Execute `terraform apply`.  
+
+
+## Known risks and considerations for upgrading Kubernetes
+- Make sure that Vantiq supports the new version of kubernetes before upgrading. Please contact Vantiq Support in advance.  
+- When upgrading the Vantiq Node pool (nodegroup) and the mongodb Node pool (nodegroup) at the same time, there will be a timing when the vantiq pod will not start, which may cause a suspension of the vantiq service. Do NOT do each at the same time.  
+- Effective from AKS 1.19, **containerd** is used instead of docker. There are two issues regarding the metrics collection done by Vantiq. One is that in System Admin Grafana, Influxdb's Measurements for CPU for Vantiq Resources and MongoDB Monitoring Dashboard have been changed, so the dashboard's query parameters need to be updated (attached separately). Another is that the connection to "docker.sock" on "ds/telegraf-ds" is continuously failing. This has not been resolved as of the update of this document, and is an operational workaround.  
```

Comment:
```suggestion
- Effective from AKS 1.19, **containerd** is used instead of docker. There are two issues regarding the metrics collection done by Vantiq. One is that in System Admin Grafana, Influxdb's Measurements for CPU for Vantiq Resources and MongoDB Monitoring Dashboard have been changed, so the dashboard's query parameters need to be updated (attached separately). Another is that the connection to "docker.sock" on "ds/telegraf-ds" is continuously failing. This has not been resolved as of the update of wiriting this article, and needs an operational workaround.  
```